CourseManager/Views/CoursesView.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  ContentView.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-09-29.
+//
+
+import FirebaseCore
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import SwiftUI
+
+struct CoursesView: View {
+    @State private var showingSheet = false
+    @State private var selectedCourseID = ""
+    
+    @ObservedObject private var viewModel = CoursesViewModel()
+
+    var body: some View {
+        NavigationView {
+            listOfCourses
+                .onAppear {
+                    self.viewModel.fetchData()
+                }
+        
+                .toolbar {
+                    Button {
+                        showingSheet.toggle()
+                    } label: {
+                        Image(systemName: "plus")
+                            .font(.title2)
+                    }.sheet(isPresented: $showingSheet) {
+                        AddOrEditCourseSheet(course: CourseInformation(id: "", name: "", url: "", gitHubURL: "", status: .todo, dateOfCompletion: Date(), comments: ""))
+                    }
+                }
+                .navigationTitle("Courses")
+        }
+    }
+    
+    var listOfCourses: some View {
+        List {
+            ForEach(viewModel.courses) { course in
+               
+                NavigationLink {
+                    CourseDetailsView(courseID: course.id!)
+                } label: {
+                    HStack {
+                        Text(course.name)
+                        Spacer()
+                        Group {
+                            if course.status == .complete {
```

Comment:
This one isn't right, read about structural identity in SwiftUI. Better to add icon, color to the extension of status and use it in Image

Reply:
Makes sense! will read up and update.

Reply:
Ok so I removed the if else statement and used a ternary operator with 3 conditions. Also I made the extension to the Status enum  as suggested to return an image based on the case. I couldn't add color to this as I needed to use the modifier foregroundStyle instead of foregroundColor to achieve 2 colors in my icons. foregroundStyle is of type 'some View ' and not image and I got a bunch of errors. so to work around this I used another turnery operator in the CoursesView as shown below. I hope this makes sense: 

enum Status: String, Codable, Equatable, CaseIterable {
    case todo = "To Do"
    case inProgress = "In Progress"
    case complete = "Complete"

    var localizedName: LocalizedStringKey { LocalizedStringKey(rawValue) }
}

extension Status {
    var image: Image {
        switch self {
        case .todo: return Image(systemName: "rectangle")
            .symbolRenderingMode(.palette)
        case .inProgress: return Image(systemName: "rectangle.lefthalf.inset.filled")
            .symbolRenderingMode(.palette)
        case .complete: return Image(systemName: "rectangle.inset.filled")
        }
    }
}
//////AND  in the CoursesView///
Group {
                            course.status == .complete ? Status.complete.image : course.status == .inProgress ? Status.inProgress.image : Status.todo.image
                        }
                        .foregroundStyle(course.status == .todo ? Color.black : Color.green, Color.black)